Bottom: 2070b40f2d9a6404eedbdf781749b1db3c0b0077
Top:    7a6cfe9b79c42b883b53b6c918eab643fe5a8687
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-23 17:31:50 +0000

Fix netlist selection to save undo properly


---

diff --git a/src/hid/gtk/gui-netlist-window.c b/src/hid/gtk/gui-netlist-window.c
index 689c5f0..4fd6fa8 100644
--- a/src/hid/gtk/gui-netlist-window.c
+++ b/src/hid/gtk/gui-netlist-window.c
@@ -516,14 +516,14 @@ netlist_select_cb (GtkWidget * widget, gpointer data)
     node_selected_net = NULL;
 
   InitConnectionLookup ();
-  ResetConnections (false, false);
+  ResetConnections (true, true);
 
   for (i = selected_net->EntryN, entry = selected_net->Entry; i; i--, entry++)
     if (SeekPad (entry, &conn, false))
       RatFindHook (conn.type, conn.ptr1, conn.ptr2, conn.ptr2, true);
 
   SelectConnection (select_flag);
-  ResetConnections (false, false);
+  ResetConnections (true, true);
   FreeConnectionLookupMemory ();
   IncrementUndoSerialNumber ();
   Draw ();
diff --git a/src/hid/lesstif/netlist.c b/src/hid/lesstif/netlist.c
index c3c522f..69cb6f2 100644
--- a/src/hid/lesstif/netlist.c
+++ b/src/hid/lesstif/netlist.c
@@ -136,14 +136,14 @@ nbcb_select_common (LibraryMenuTypePtr net, int pos, int select_flag)
   int i;
 
   InitConnectionLookup ();
-  ResetConnections (false, false);
+  ResetConnections (true, true);
 
   for (i = net->EntryN, entry = net->Entry; i; i--, entry++)
     if (SeekPad (entry, &conn, false))
       RatFindHook (conn.type, conn.ptr1, conn.ptr2, conn.ptr2, true);
 
   SelectConnection (select_flag);
-  ResetConncetions (false, false);
+  ResetConncetions (true, true);
   FreeConnectionLookupMemory ();
   IncrementUndoSerialNumber ();
   Draw ();
